### ISSUE TITLE
Non-existent URL returns HTTP 200 with empty body

### ISSUE THREAD

**Issue by fedys** [cross-solution/YAWIK#208]
If you enter for example http://yawik.org/demo/en/non-existent you get HTTP 200 with empty body. Expected result would be HTTP 404 with simple body describing non-existent page.

**Comment by TiSiE** [comment]
Indeed. And the cause is the [InjectSubnavigationListener](https://github.com/cross-solution/YAWIK/blob/develop/module/Settings/src/Settings/Listener/InjectSubNavigationListener.php), which throws the following exception while the error template is rendered:

EXCEPTION Could not get a Route Match. route is invalid. in /data/www/clients/client1/web1/YAWIK/module/Settings/src/Settings/Listener/InjectSubNavigationListener.php on line 48

**Comment by fedys** [comment]
I can try to fix it, ok?

**Comment by TiSiE** [comment]
Yes! Just silently exists instead of throwing an exception should be sufficient.

Please push to a new branch, I will then review and merge.

**Comment by fedys** [comment]
It's a deal

**Comment by fedys** [comment]
I've pushed 'http-404-fix' branch. I chose another approach because there were multiple places to fix missing RouteMatch during displaying the 404 page. So I create a dummy 'no-route' route on-fly in case of no route error. This way all templates/listeners can work as expected and there is no need to fix non-existing RouteMatch. I've already used this approach in several projects without any side-effects.
